### PR TITLE
Updated iterable dataset with len warning message

### LINKED ISSUE #6966: Multi-processing with IterableDataset Warning
Hi there. This is a report for a potential typo, and a related question.

The warning message given when using an IterableDataset is as follows: 

https://github.com/PyTorchLightning/pytorch-lightning/blob/049006a59cf62d3539aaf19037aca1caac98abfd/pytorch_lightning/utilities/data.py#L40-L45

Is the part `e.g. batch size > 1` should be `e.g. num_workers > 1`?

Is this warning about what [IterableDataset doc](https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset) explains about configuring the workers? If I write my IterableDataset similarly to the example provided in the doc and also define a `__len__` function, will there be any problems due to using PL?

Thanks.

### PR DESCRIPTION
**Pull request by ozen**
Based on pytorch docs about dataloader length and iterable datasets.

## What does this PR do?

Fixes #6966

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/utilities/data.py [hunk 1]
```diff
@@ -40,8 +40,9 @@ def has_len(dataloader: DataLoader) -> bool:
     if has_len and has_iterable_dataset(dataloader):
         rank_zero_warn(
             'Your `IterableDataset` has `__len__` defined.'
-            ' In combination with multi-processing data loading (e.g. batch size > 1),'
-            ' this can lead to unintended side effects since the samples will be duplicated.'
+            ' In combination with multi-process data loading (when num_workers > 1),'
+            ' `__len__` could be inaccurate if each worker is not configured independently'
+            ' to avoid having duplicate data.'
```
Comment (carmocca) [inline pytorch_lightning/utilities/data.py:45]: I don't quite get the message. It checks that `__len__` is not defined but then says `__len__` could be inaccurate...

How can it not be defined and inaccurate at the same time?

Comment (ozen) [reply pytorch_lightning/utilities/data.py:45]: It displays the warning if len is defined, not the other way around.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/utilities/data.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/data.py:45]: oops im blind 🙈

**Review by carmocca — APPROVED** [review]
